pam_tacplus.c

```diff
@@ -654,33 +654,29 @@ int pam_sm_acct_mgmt(pam_handle_t * pamh, int flags, int argc,
 
 	attr = arep.attr;
 	while (attr != NULL) {
-		char attribute[attr->attr_len];
-		char value[attr->attr_len];
-		char *sep;
-
-		sep = index(attr->attr, '=');
-		if (sep == NULL)
-			sep = index(attr->attr, '*');
-		if (sep != NULL) {
-			bcopy(attr->attr, attribute, attr->attr_len - strlen(sep));
-			attribute[attr->attr_len - strlen(sep)] = '\0';
-			bcopy(sep, value, strlen(sep));
-			value[strlen(sep)] = '\0';
+		size_t len = strcspn(attr->attr, "=*");
+		if (len < attr->attr_len) {
+			char avpair[attr->attr_len+1];
+			memcpy(avpair, attr->attr, attr->attr_len+1); /* Also copy terminating NUL */
```

Comment:
We started out using `bcopy()` so let's stick to that.

Reply:
I used memcpy() because it's my understanding that bcopy() is deprecated (marked as LEGACY in
POSIX.1-2001). I will submit a change using bcopy() in just a few minutes.